backend/dal/Services/Admin/Concrete/UserService.cs

```diff
@@ -150,6 +150,65 @@ public override void Remove(User entity)
             base.Remove(user);
         }
 
+        /// <summary>
+        /// Updates the specified access request in the datasource. if the request is granted, update the associated user as well.
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <exception type="KeyNotFoundException">Entity does not exist in the datasource.</exception>
+        /// <returns></returns>
+        public AccessRequest Update(AccessRequest entity)
+        {
+            if(entity.UserId == null)
+            {
+                throw new ArgumentNullException("userId");
+            }
+            entity.ThrowIfNull(nameof(entity));
+            this.User.ThrowIfNotAuthorized(Permissions.SystemAdmin, Permissions.AgencyAdmin);
+
+            var accessRequest = this.Context.AccessRequests
+                .Include(p => p.Agencies)
+                .ThenInclude(p => p.Agency)
+                .Include(p => p.Roles)
+                .ThenInclude(p => p.Role)
+                .Include(p => p.User)
+                .FirstOrDefault(a => a.Id == entity.Id) ?? throw new KeyNotFoundException();
+            if(!accessRequest.IsGranted != true && entity.IsGranted == true && accessRequest.UserId.HasValue)
+            {
+                User user = Get(accessRequest.UserId.Value);
+                entity.Agencies.ForEach((accessRequestAgency) =>
+                {
+                    if (!user.Agencies.Select(a => a.AgencyId).Contains(accessRequestAgency.AgencyId))
+                    {
+                        user.Agencies.Add(new UserAgency()
+                        {
+                            User = user,
+                            Agency = this.Context.Agencies.Find(accessRequestAgency.AgencyId)
+                        });
+                    }
+                });
+                entity.Roles.ForEach((accessRequestRole) =>
+                {
+                    if (!user.Roles.Select(r => r.RoleId).Contains(accessRequestRole.RoleId))
+                    {
+                        user.Roles.Add(new UserRole()
+                        {
+                            User = user,
+                            Role = this.Context.Roles.Find(accessRequestRole.RoleId)
+                        });
+                    }
+                });
+                Update(user);
```

Comment:
This update to the user will occur in its own transaction.  You need to use `UpdateOne(user)`.  

Think I need a better naming convention for those methods...